### ISSUE TITLE
False positive `no-value-for-parameter` for Qt methods

### ISSUE THREAD

**Issue by jacobtylerwalls** [PyCQA/pylint#6464]
### Bug description

Reproducer thx to @The-Compiler 

```python
from PyQt5.QtCore import QTimer
timer = QTimer()
timer.timeout.connect(lambda: None)
```

### Configuration

```ini
--extension-pkg-whitelist=PyQt5
```

### Command used

```shell
pylint a.py --extension-pkg-whitelist=PyQt5
```

### Pylint output

```shell
a.py:4:0: C0305: Trailing newlines (trailing-newlines)
a.py:1:0: C0114: Missing module docstring (missing-module-docstring)
a.py:3:0: E1120: No value for argument 'slot' in method call (no-value-for-parameter)
```

### Expected behavior

a.py:4:0: C0305: Trailing newlines (trailing-newlines)
a.py:1:0: C0114: Missing module docstring (missing-module-docstring)

### Pylint version

```shell
2.14 dev, essentially since astroid changes in https://github.com/PyCQA/astroid/pull/1505
```
